File: sdk/samples/WCFSamples/TechnologySamples/Scenario/Router/VB/service-echo/EchoService.vb
' Copyright (C) 2003-2005 Microsoft Corporation, All rights reserved.

Imports System
Imports System.ServiceModel
Imports System.ServiceModel.Dispatcher
Imports System.Diagnostics
Imports System.Configuration

Namespace Microsoft.ServiceModel.Samples

    <ServiceContract([Namespace]:="http://Microsoft.ServiceModel.Samples/")> _
    Public Interface IEchoService

        <OperationContract(IsOneWay:=False)> _
        Function Echo(ByVal message As String) As String

    End Interface

    <ServiceBehavior(InstanceContextMode:=InstanceContextMode.[Single])> _
    Public Class EchoService
        Implements IEchoService

        Private Function Echo(ByVal message As String) As String Implements IEchoService.Echo

            Return message

        End Function

    End Class

    Public Class ServiceDriver

        Public Shared Sub Main(ByVal args As String())

            Dim serviceHost As New ServiceHost(GetType(EchoService))
            serviceHost.Open()

            For i As Integer = 0 To serviceHost.ChannelDispatchers.Count - 1

                Dim channelDispatcher As ChannelDispatcher = TryCast(serviceHost.ChannelDispatchers(i), ChannelDispatcher)
                If channelDispatcher IsNot Nothing Then

                    For j As Integer = 0 To channelDispatcher.Endpoints.Count - 1

                        Dim endpointDispatcher As EndpointDispatcher = channelDispatcher.Endpoints(j)
                        Console.WriteLine("Listening on " & endpointDispatcher.EndpointAddress.ToString() & "...")

                    Next

                End If

            Next

            Console.WriteLine()
            Console.WriteLine("Press Enter to exit...")
            Console.ReadLine()

        End Sub

    End Class

End Namespace

File: sdk/samples/WCFSamples/TechnologySamples/Scenario/Router/CS/service-echo/EchoService.cs
// ----------------------------------------------------------------------------
// Copyright (C) 2003-2005 Microsoft Corporation, All rights reserved.
// ----------------------------------------------------------------------------
namespace Microsoft.ServiceModel.Samples
{
    using System;
    using System.ServiceModel;
	using System.ServiceModel.Dispatcher;
    using System.Diagnostics;
    using System.Configuration;

    [ServiceContract(Namespace = "http://Microsoft.ServiceModel.Samples/")]
    public interface IEchoService
    {
        [OperationContract(IsOneWay = false)]
        string Echo(string message);
    }

    [ServiceBehavior(InstanceContextMode = InstanceContextMode.Single)]
    public class EchoService : IEchoService
    {
        string IEchoService.Echo(string message)
        {
            return message;
        }
    }

    public class ServiceDriver
    {
        public static void Main(string[] args)
        {
            ServiceHost serviceHost = new ServiceHost(typeof(EchoService));
            serviceHost.Open();

            for (int i = 0; i < serviceHost.ChannelDispatchers.Count; i++)
            {
                ChannelDispatcher channelDispatcher = serviceHost.ChannelDispatchers[i] as ChannelDispatcher;
                if (channelDispatcher != null)
                {
                    for (int j = 0; j < channelDispatcher.Endpoints.Count; j++)
                    {
                        EndpointDispatcher endpointDispatcher = channelDispatcher.Endpoints[j];
                        Console.WriteLine("Listening on " + endpointDispatcher.EndpointAddress + "...");
                    }
                }
            }

            Console.WriteLine();
            Console.WriteLine("Press Enter to exit...");
            Console.ReadLine();
        }
    }
}
